sources/file-vault/src/main/java/io/smallrye/config/source/filevault/FileVaultConfigSourceInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package io.smallrye.config.source.filevault;
+
+import static io.smallrye.common.expression.Expression.Flag.LENIENT_SYNTAX;
+import static io.smallrye.common.expression.Expression.Flag.NO_SMART_BRACES;
+import static io.smallrye.common.expression.Expression.Flag.NO_TRIM;
+
+import java.util.function.BiConsumer;
+
+import io.smallrye.common.expression.Expression;
+import io.smallrye.common.expression.ResolveContext;
+import io.smallrye.config.ConfigSourceInterceptor;
+import io.smallrye.config.ConfigSourceInterceptorContext;
+import io.smallrye.config.ConfigValue;
+
+public class FileVaultConfigSourceInterceptor implements ConfigSourceInterceptor {
+    private static final long serialVersionUID = -45605050209974890L;
+
+    @Override
+    public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
+        ConfigValue configValue = context.proceed(name);
+
+        if (configValue == null) {
+            return null;
+        }
+
+        Expression expression = Expression.compile(configValue.getValue(), LENIENT_SYNTAX, NO_TRIM, NO_SMART_BRACES);
+        String expanded = expression.evaluate(new BiConsumer<ResolveContext<RuntimeException>, StringBuilder>() {
```

Comment:
I believe that having multiple property expanders in series will cause subtle syntactic problems, particularly with nested expressions but probably some other cases as well (default values maybe?). This is why I proposed using the double-colon feature on the existing property expander, and then adding a separate hook to register handlers for it: it avoids any syntactic problems that can occur with multiple series-style interpolations (which should be regarded as an anti-pattern IMO).

Reply:
Sure, all of this can be changed. The idea was to ensure that it could be implemented with the interceptor approach, and I didn't want to start adding new APIs without some validation first.

Separate handlers will require access to the ConfigContext, and we need to figure out when they should be initialized.